dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/TextCompletionExtensions.cs

```diff
@@ -54,7 +54,6 @@ await foreach (var word in completionResult.GetCompletionStreamingAsync(cancella
             {
                 yield return word;
             }
-            yield break;
```

Comment:
This extension method is intended for single results only.

See the comment on the method.
`This extension does not support multiple prompt results (Only the first will be returned)`

Without the `break` this will yield all the multiple responses as one big text making the result indistinguishable and confusing.

Reply:
OK I believe I got it right this time. 
I restored the yield break while keeping the asynchronous streaming behavior, which was hard to get right.

- I used Channels to enabled on demand streaming iteration in the Result class in a simple implementation,
- I added yet another asynchronous Task to process websockets in order to be able to yield return the first incomplete StreamingResult after the first websocket message chunk is received, while preventing the "yield break" from this extension method to freeze subsquent websocket processing, which was leading to a deadlock where the iterator consumer would wait for the "stream_end" websocket message to terminate and the websocket processing would keep frozen waiting for the iterator to end to resume websocket processing. 
That dead lock took me sometime to understand and properly fix.

That is to say: that yield break puts a hard constraint on the way we can implement streaming completion. I wonder if a simple "break;" would have been gentler.
In any case, I hope my code can help other move forward with supporting websocket servers.

Reply:
@RogerBarreto 
I did some more work, with comments about them below. Maybe we can resolve that earlier conversation.